nvflare/private/fed/app/simulator/simulator_runner.py

```diff
@@ -0,0 +1,325 @@
+# Copyright (c) 2021-2022, NVIDIA CORPORATION.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import logging
+import os
+import shutil
+import sys
+import tempfile
+import threading
+import time
+from concurrent.futures import ThreadPoolExecutor
+
+from nvflare.apis.event_type import EventType
+from nvflare.apis.fl_component import FLComponent
+from nvflare.apis.fl_constant import FLContextKey, MachineStatus, WorkspaceConstants
+from nvflare.apis.job_def import ALL_SITES, JobMetaKey
+from nvflare.apis.workspace import Workspace
+from nvflare.fuel.hci.server.authz import AuthorizationService
+from nvflare.fuel.hci.zip_utils import convert_legacy_zip, split_path, unzip_all_from_bytes, zip_directory_to_bytes
+from nvflare.fuel.sec.audit import AuditService
+from nvflare.private.defs import AppFolderConstants, EngineConstant
+from nvflare.private.fed.app.deployer.simulator_deployer import SimulatorDeploy
+from nvflare.private.fed.client.client_json_config import ClientJsonConfigurator
+from nvflare.private.fed.client.client_run_manager import ClientRunManager
+from nvflare.private.fed.client.client_runner import ClientRunner
+from nvflare.private.fed.client.client_status import ClientStatus
+from nvflare.private.fed.server.job_meta_validator import JobMetaValidator
+from nvflare.private.fed.simulator.simulator_client_app_runner import SimulatorServerAppRunner
+from nvflare.security.security import EmptyAuthorizer
+
+
+class SimulatorRunner(FLComponent):
```

Comment:
Why do we want to inherit FLComponent?
This SimulatorRunner is not part of the system and it does not have "handle_event".

If we need logger maybe we just add logger in the ``__init__`` function

Reply:
Make the SimulatorRunner a FLComponent enable it the ability to handle events, or fire events if needed. Actually the current client_runner.fire_event(EventType.SWAP_OUT, fl_ctx) needs to change to self.fire_event(EventType.SWAP_OUT, fl_ctx).